src/main/java/seedu/address/logic/parser/AddNoteCommandParser.java

```diff
@@ -34,15 +35,21 @@ public AddNoteCommand parse(String args) throws ParseException {
 
         IdentityCardNumber ic;
         try {
-            String note = argMultimap.getValue(PREFIX_NOTE).orElse("");
+            String note = "";
+            boolean isReplace = false;
 
-            boolean isReplace;
-            if (note.contains("-replace")) {
+            if (argMultimap.getValue(PREFIX_FLAG).isPresent()) {
+                int startIndex = args.indexOf(PREFIX_NOTE.getPrefix()) + PREFIX_NOTE.getPrefix().length();
+                int endIndex = args.indexOf(PREFIX_FLAG.getPrefix());
+                if (endIndex == -1) {
```

Comment:
what is -1? would you want to leave a comment explaining to help explain?

Reply:
endIndex == -1 is used to check if the endIndex variable has been updated from its initial value, which is -1 when PREFIX_FLAG is not found in the args string.
But then I realised that its not going to be needed because in the outer if statement I am already checking if the flag is present. 
So I removed it!